### DISCUSSION TITLE
Database search (express.js)

### DISCUSSION THREAD

**Discussion by scriptyyy13** [deta/cloud-docs#508] [Help] [answered]
Is there any normal way to do a deta base search on an incomplete name? I tried to make such a search, but it did not work and gave the result only when the value was fully entered.

CODE:
```js
app.post('/search', async (req, res) => {
    const { items } = await an.fetch({"name": req.body.name});
    if (items) { res.redirect('./anime/'+items[0].key); } else { res.redirect('./anime/2'); }
});

**Comment by Sokhavuth** [comment, marked as answer]
Maybe you could do like this {"name?contains": req.body.name}. It works for string and array.
[https://docs.deta.sh/docs/base/queries](https://docs.deta.sh/docs/base/queries)

**Reply by scriptyyy13** [reply]
thanks
